pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1334,33 +1335,36 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
             log.debug("[{}] Checking replication status", name);
         }
 
-        CompletableFuture<Policies> policiesFuture = brokerService.pulsar().getPulsarResources()
-                .getNamespaceResources()
-                .getPoliciesAsync(TopicName.get(topic).getNamespaceObject())
-                .thenCompose(optPolicies -> {
-                            if (!optPolicies.isPresent()) {
-                                return FutureUtil.failedFuture(
-                                        new ServerMetadataException(
-                                                new MetadataStoreException.NotFoundException()));
-                            }
+        CompletableFuture<Set<String>> replicationClustersFuture = brokerService.pulsar()
```

Comment:
The checkReplication is call by `onPoliciesUpdate`, which is listen namespace level policy update. For topic level policy update listener, we should deal with the topic policy in `onUpdate()` method.

Reply:
From the discussion, my implement and your suggestion both can realize this.

Reply:
Current implementation relies on `replicationPolicyCheckDurationSeconds = 600` to check replication policy update. We can merge this PR first, and consider use another PR to update the replication policy by listener.